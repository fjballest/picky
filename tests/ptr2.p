/*
!../8.pc   ptr2.p && ../8.pi  p.out >[2=1]
 */
program ptr2;

types:
	Pint =  ^int;

procedure main()
	p: Pint;
	q: Pint;
{
	new(p);
	p^ = 3;
	new(q);
	dispose(q);
	q = p;
	stack();
	data();
	writeln(q^);
	dispose(p);
	stack();
	data();
}
