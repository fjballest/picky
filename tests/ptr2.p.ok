stack trace at:
main() pid 0	pc 0x000021 ptr2.p:18
	local variables:
	p = 0x401aba0
	q = 0x401aba0

global variables:
Maxint = 2147483647
Minint = 0
Maxchar = '每'
Minchar = '\000'
Eol = '\n'
Eof = '每'
Tab = '\t'
Esc = '\033'
Nul = '\000'
stdin = <stdin>
stdout = <stdout>
heap:
0x401aba0 = 3
3
stack trace at:
main() pid 0	pc 0x000038 ptr2.p:22
	local variables:
	p = 0x401aba0
	q = 0x401aba0

global variables:
Maxint = 2147483647
Minint = 0
Maxchar = '每'
Minchar = '\000'
Eol = '\n'
Eof = '每'
Tab = '\t'
Esc = '\033'
Nul = '\000'
stdin = <stdin>
stdout = <stdout>
heap:
