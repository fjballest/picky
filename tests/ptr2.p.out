stack trace at:
main() pid 0	pc 0x000021 ptr2.p:18
	local variables:
	p = 0x401ac38
	q = 0x401ac38

global variables:
Maxint = 2147483647
Minint = -2147483646
Maxchar = '每'
Minchar = '\000'
Eol = '\n'
Eof = '每'
Tab = '\t'
Esc = '\033'
Nul = '\000'
stdin = <stdin>
stdout = <stdout>
heap:
0x401ac38 = 3
3
stack trace at:
main() pid 0	pc 0x000038 ptr2.p:22
	local variables:
	p = 0x401ac38
	q = 0x401ac38

global variables:
Maxint = 2147483647
Minint = -2147483646
Maxchar = '每'
Minchar = '\000'
Eol = '\n'
Eof = '每'
Tab = '\t'
Esc = '\033'
Nul = '\000'
stdin = <stdin>
stdout = <stdout>
heap:
